packages/clerk-js/src/core/clerk.ts

```diff
@@ -124,19 +132,29 @@ export default class Clerk implements ClerkInterface {
     return this.#isReady;
   }
 
-  public constructor(frontendApi: string) {
-    if (!frontendApi) {
-      clerkErrorNoFrontendApi();
-    }
-    if (!validateFrontendApi(frontendApi)) {
-      clerkErrorInvalidFrontendApi();
-    }
+  public constructor(key: string) {
+    key = (key || '').trim();
 
-    this.frontendApi = frontendApi;
+    if (isLegacyFrontendApiKey(key)) {
+      if (!validateFrontendApi(key)) {
+        errorThrower.throwInvalidFrontendApiError({ key });
+      }
 
-    this.#instanceType = isDevOrStagingUrl(this.frontendApi) ? 'development' : 'production';
-    this.#fapiClient = createFapiClient(this);
+      this.frontendApi = key;
+      this.#instanceType = isDevOrStagingUrl(this.frontendApi) ? 'development' : 'production';
+    } else {
+      const publishableKey = parsePublishableKey(key);
 
+      if (!publishableKey) {
+        errorThrower.throwInvalidPublishableKeyError({ key });
+      }
+
+      const { frontendApi, instanceType } = publishableKey as PublishableKey;
+
+      this.frontendApi = frontendApi;
+      this.#instanceType = instanceType;
```

Comment:
🙃 Is this required after all? If not, feel free to remove.

Reply:
Yeap actually we used it in some places to handle differently some flows (e.g. dev browser, build redirect urls)